### ISSUE TITLE
absolute_file_name/3 crash

### ISSUE THREAD

**Issue by pmoura** [infradig/trealla#202]
```text
$ tpl
Trealla Prolog (c) Infradig 2020, v1.8.66
?- absolute_file_name('$HOME', Path, [relative_to('/Users/pmoura')]).
tpl(94505,0x11b74fe00) malloc: *** error for object 0x7fbf4400e8a0: pointer being freed was not allocated
tpl(94505,0x11b74fe00) malloc: *** set a breakpoint in malloc_error_break to debug
Abort trap: 6
```

The crash seems to be caused by the atom argument for the `relative_to/1` option. Given that an atom is accepted for the first argument, it would be nice and consistent to also accept an atom for this option.

**Comment by infradig** [comment]
Not actually related to that, but fixed the bad free #203.

**Comment by pmoura** [comment]
No longer a crash:

```text
$ tpl
Trealla Prolog (c) Infradig 2020, v1.8.66-3-gc4128
?- absolute_file_name('$HOME', Path, [relative_to('/Users/pmoura')]).
Path = '/Users/pmoura/$HOME'.
```

Thanks for the fix.

P.S. The result looks wrong but that's just because the `expand` option defaults to `false`.
